pkg/clustersync/clusterSync.go

```diff
@@ -62,17 +62,17 @@ func syncClusters(ctx context.Context) {
 	// Create handlers for events
 	handlers := cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			klog.V(4).Info("clusterWatch: AddFunc for ", obj.(*unstructured.Unstructured).GetKind())
-			processClusterUpsert(obj)
+			klog.V(4).Info("AddFunc for ", obj.(*unstructured.Unstructured).GetKind())
+			processClusterUpsert(ctx, obj)
 		},
 		UpdateFunc: func(prev interface{}, next interface{}) {
-			klog.V(4).Info("clusterWatch: UpdateFunc for", next.(*unstructured.Unstructured).GetKind())
-			processClusterUpsert(next)
+			klog.V(4).Info("UpdateFunc for", next.(*unstructured.Unstructured).GetKind())
+			processClusterUpsert(ctx, next)
+		},
+		DeleteFunc: func(obj interface{}) {
+			klog.V(4).Info("DeleteFunc")
+			processClusterDelete(ctx, obj)
```

Comment:
This will receive delete events from both ManagedCluster and ManagedClusterInfo.  I wonder if this can cause problems or if we should limit it to only ManagedCluster.

I want to check how the API behaves. Does deleting ManagedCluster automatically deletes ManagedClusterInfo?  Is there a valid scenario where only 1 of the 2 resources is deleted.

Reply:
I also want to fin out if there's any scenario that could add resources for a cluster after we have processed the delete.
The proxy should stop forwarding requests after the cluster is deleted but I want to find out the timing of this.
Also the sync request is asynchronous and could take longer to complete. We may need to worry about canceling any activity in progress.  This will get harder with multiple replicas of the indexer. 🤔

Reply:
From @zhiweiyin318: 

> ManagedCluster is a cluster-scoped resource, ManagedClusterInfo is a namespace-scoped resource which includes the most info of ManagedCluster. ManagedClusterInfo will be deleted automatically after ManagedCluster is deleted.
> The ManagedClusterInfo will be deleted since the cluster ns will be deleted after ManagedCluster is deleted.